system/lib/wasmfs/syscalls.cpp

```diff
@@ -290,6 +286,74 @@ long __syscall_fstat64(long fd, long buf) {
   return __WASI_ERRNO_SUCCESS;
 }
 
+long __syscall_stat64(long path, long buf) {
+  auto pathParts = splitPath((char*)path);
+
+  if (pathParts.empty()) {
+    return -EINVAL;
+  }
+
+  auto base = pathParts.back();
+
+  long err;
+  auto parentDir = getDir(pathParts.begin(), pathParts.end() - 1, err);
+
+  // Parent node doesn't exist.
+  if (!parentDir) {
+    return err;
+  }
+
+  auto lockedParentDir = parentDir->locked();
+
+  // TODO: In future PR, edit function to just return the requested file instead
+  // of having to first obtain the parent dir.
+  auto curr = lockedParentDir.getEntry(base);
+  if (curr) {
+    return __stat64(curr, buf);
+  } else {
+    return -ENOENT;
+  }
+}
+
+long __syscall_lstat64(long path, long buf) {
+  // TODO: When symlinks are introduced, lstat will return information about the
+  // link itself rather than the file it refers to.
+  auto pathParts = splitPath((char*)path);
+
+  if (pathParts.empty()) {
+    return -EINVAL;
+  }
+
+  auto base = pathParts.back();
+
+  long err;
+  auto parentDir = getDir(pathParts.begin(), pathParts.end() - 1, err);
+
+  // Parent node doesn't exist.
+  if (!parentDir) {
+    return err;
+  }
+
+  auto lockedParentDir = parentDir->locked();
+
+  auto curr = lockedParentDir.getEntry(base);
+  if (curr) {
+    return __stat64(curr, buf);
+  } else {
+    return -ENOENT;
+  }
+}
```

Comment:
Aside from comments, the lstat syscall code looks identical to the stat syscall code, so more could be shared between them I think.

Reply:
We can just call stat from lstat for now 👍